### PR TITLE
Support optional/configurable IAMEndpoint for Minio Client (#32581)

### PR DESCRIPTION
**Pull request by mowoc-ocp**
Targeting issue #32271

This modification allows native Kubernetes + AWS (EKS) authentication with the Minio client, to Amazon S3 using the IRSA role assigned to a Service account by replacing the hard coded reference to the `DefaultIAMRoleEndpoint` with an optional configurable endpoint.

Internally, Minio's `credentials.IAM` provider implements a discovery flow for IAM Endpoints if it is not set. 

For backwards compatibility: 
- We have added a configuration mechanism for an `IamEndpoint` to retain the unit test safety in `minio_test.go`. 
- We believe existing clients will continue to function the same without needing to provide a new config property since the internals of Minio client also often resolve to the `http://169.254.169.254` default endpoint that was being hard coded before

To test, we were able to build a docker image from source and, observe it choosing the expected IAM endpoint, and see files uploaded via the client.

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]
Looks good.

by the way, "app.example.ini" and related documents https://gitea.com/gitea/docs/pulls also need to update 🙏

File: modules/setting/storage.go
Comment (lunny) [inline modules/setting/storage.go:46]: So maybe it should have a default value `credentials.DefaultIAMRoleEndpoint`?

Comment (wxiaoguang) [reply modules/setting/storage.go:46]: No need.

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> by the way, "app.example.ini" and related documents https://gitea.com/gitea/docs/pulls also need to update 🙏

Wait for config example update before merge. If there is any difficulty, maintainers could also help.

**Comment by mowoc-ocp** [comment]
@wxiaoguang thank you for the feedback, I'll take care of the documentation PR today!

**Comment by mowoc-ocp** [comment]
Added the requested updates for `app.example.ini` and opened https://gitea.com/gitea/docs/pulls/103

Let me know if there is any feedback on documentation structure or anywhere I missed that I also need to update. Thanks!

**Pull request merged**
